pages/index.js

```diff
@@ -56,25 +56,27 @@ export default function HomePage({ onSubmit, setIsEditMode }) {
   };
 
   function getFilteredActivities() {
-    if (!isLoading && activities) {
-      let filtered = activities;
-
-      if (authorFilter) {
-        filtered = filtered.filter(
-          (activity) =>
-            activity.author.toLowerCase() === authorFilter.toLowerCase()
-        );
-      }
+    if (isLoading || !activities) {
+      return [];
+    }
 
-      if (categoryFilter) {
-        filtered = filtered.filter(
-          (activity) =>
-            activity.category.toLowerCase() === categoryFilter.toLowerCase()
-        );
-      }
+    let filtered = activities;
```

Comment:
If you rename this variable to filteredActivities it would be clearer what it contains.